src/core/Akka.Remote.Tests.MultiNode/RemoteReDeploymentSpec.cs

```diff
@@ -0,0 +1,307 @@
+    {
+        public RemoteReDeploymentSpecConfig()
+        {
+            First = Role("first");
+            Second = Role("second");
+
+            CommonConfig = DebugConfig(true).WithFallback(ConfigurationFactory.ParseString(@"
+    akka.remote.transport-failure-detector {
+         threshold=0.1
+         heartbeat-interval=0.1s
+         acceptable-heartbeat-pause=1s
+       }
+       akka.remote.watch-failure-detector {
+         threshold=0.1
+         heartbeat-interval=0.1s
+         acceptable-heartbeat-pause=2.5s
+       }"));
+
+            DeployOn(Second, "/parent/hello.remote = \"@first@\"");
+
+            TestTransport = true;
+        }
+
+        public RoleName First { get; }
+        public RoleName Second { get; }
+    }
+
+
+    public abstract class RemoteReDeploymentSpec : MultiNodeSpec
+    {
+        private readonly RemoteReDeploymentSpecConfig _config;
+        private readonly Func<RoleName, string, IActorRef> _identify;
+
+        protected RemoteReDeploymentSpec() : this(new RemoteReDeploymentSpecConfig())
+        {
+        }
+
+        protected RemoteReDeploymentSpec(RemoteReDeploymentSpecConfig config) : base(config)
+        {
+            _config = config;
+        }
+
+        protected abstract bool ExpectQuarantine { get; }
+        protected abstract TimeSpan SleepAfterKill { get; }
+
+        protected override int InitialParticipantsValueFactory
+        {
+            get { return Roles.Count; }
+        }
+
+        [MultiNodeFact]
+        public void RemoteReDeployment_must_terminate_the_child_when_its_parent_system_is_replaced_by_a_new_one()
+        {
+            var echo = Sys.ActorOf(EchoProps(TestActor), "echo");
+            EnterBarrier("echo-started");
+
+            RunOn(() =>
+            {
+                Sys.ActorOf(Props.Create(() => new Parent()), "parent")
+                    .Tell(new ParentMessage(Props.Create(() => new Hello()), "hello"));
+
+                ExpectMsg<string>("HelloParent", TimeSpan.FromSeconds(15));
+            }, _config.Second);
+
+            RunOn(() =>
+            {
+                ExpectMsg<string>("PreStart", TimeSpan.FromSeconds(15));
+                
+            }, _config.First);
+
+            EnterBarrier("first-deployed");
+
+            RunOn(() =>
+            {
+                TestConductor.Blackhole(_config.Second, _config.First, ThrottleTransportAdapter.Direction.Both)
+                    .Wait();
+                TestConductor.Shutdown(_config.Second, true).Wait();
+                if (ExpectQuarantine)
+                {
+                    
+                    Within(SleepAfterKill, () => 
+                    {
+                        ExpectMsg("PostStop");
+                        //need to pad the timing here, since `ExpectNoMsg` will wait until exactly SleepAfterKill and fail the spec
+                        ExpectNoMsg(Remaining - TimeSpan.FromSeconds(0.2));
```

Comment:
Found a fun bug with the normal `Akka.TestKit` here which I will create an issue for. This was my workaround.